manager/integration/tests/test_basic.py

```diff
@@ -3324,28 +3326,73 @@ def test_allow_volume_creation_with_degraded_availability_error(
-    2. Deploy this StatefulSet:
+    1. create new StorageClass with numberOfReplicas=5.
+    2. Set allow-volume-creation-with-degraded-availability to True.
+    3. Deploy this StatefulSet:
        https://github.com/longhorn/longhorn/issues/2073#issuecomment-742948726
-    3. In a 1-min retry loop, Verify that all 10 volumes are healthy
-    4. Delete the StatefulSet
-    5. In a 1-min retry loop, Verify that all 10 volumes are detached
-    6. Find and delete the PVC of the 10 volumes.
-    7. In a 1-min retry loop, Verify that all 10 volumes are deleted
-    8. Make sure to delete all extra storage classes in
-       common.cleanup_client()
-    """
-    pass
+    4. In a 1-min retry loop, Verify that all 10 volumes are healthy.
+    5. Delete the StatefulSet.
+    6. In a 1-min retry loop, Verify that all 10 volumes are detached.
+    7. Find and delete the PVC of the 10 volumes.
+    8. In a 1-min retry loop, Verify that all 10 volumes are deleted.
+    9. Make sure to delete all extra storage classes in
+       common.cleanup_client().
+    """
+    storage_class['parameters']['numberOfReplicas'] = "5"
+    create_storage_class(storage_class)
+
+    setting = client.by_id_setting(common.SETTING_DEGRADED_AVAILABILITY)
+    client.update(setting, value="true")
+
+    sts_spec = statefulset['spec']
+    sts_spec['podManagementPolicy'] = "Parallel"
+    sts_spec['replicas'] = 10
+    sts_spec['volumeClaimTemplates'][0]['spec']['storageClassName'] = \
+        storage_class['metadata']['name']
+    common.create_statefulset(statefulset)
+
+    retry_count = int(60 / RETRY_INTERVAL)
+    for _ in range(common.DEFAULT_STATEFULSET_TIMEOUT):
+        try:
+            pod_info = common.get_statefulset_pod_info(core_api, statefulset)
+            break
+        except ApiException as e:
+            assert e.status == 404
+            common.DEFAULT_STATEFULSET_INTERVAL
+    assert pod_info is not None
+
+    common.wait_for_statefulset_volume_state(client, statefulset, pod_info,
+                                             common.VOLUME_FIELD_ROBUSTNESS,
+                                             common.VOLUME_ROBUSTNESS_HEALTHY,
+                                             retry_counts=retry_count)
+
+    apps_api.delete_namespaced_stateful_set(
+        name=statefulset['metadata']['name'],
+        namespace=statefulset['metadata']['namespace'],
+        body=k8sclient.V1DeleteOptions()
+    )
+
+    common.wait_for_statefulset_volume_state(client, statefulset, pod_info,
+                                             common.VOLUME_FIELD_STATE,
+                                             common.VOLUME_STATE_DETACHED,
+                                             retry_counts=retry_count)
+
+    for p in pod_info:
+        core_api.delete_namespaced_persistent_volume_claim(
+            name=p['pvc_name'], namespace='default',
+            body=k8sclient.V1DeleteOptions())
```

Comment:
Use `delete_and_wait_pvc`

Reply:
`delete_and_wait_pvc` loop through 2.5 mins to wait for pvc to get removed, this looks like it contradicts with the testing step `In a 1-min retry loop, Verify that all 10 volumes are deleted`... do we still want to change this?

Reply:
I don't think `1-min` is a hard requirement. @PhanLe1010 Is using a longer wait interval OK in this test?

Reply:
Since the original goal statement is `We want to verify that multiple volumes with degraded availability can be created, attached, detached, and deleted at nearly the *same time*`, I've updated to use `delete_and_wait_pvc` as well as with loop for 1-min retry loop.